tests/subsys/llext/simple/src/test_llext_simple.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (c) 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/kernel.h>
+#include <zephyr/llext/llext.h>
+#include <zephyr/llext/buf_loader.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(test_llext_simple);
+
+
+#ifdef CONFIG_LLEXT_STORAGE_WRITABLE
+#define LLEXT_CONST const
+#else
+#define LLEXT_CONST
+#endif
+
+struct llext_test {
+	const char *name;
+	bool try_userspace;
+	size_t buf_len;
+	LLEXT_CONST uint8_t *buf;
+};
+
+
+
+K_THREAD_STACK_DEFINE(llext_stack, 1024);
+struct k_thread llext_thread;
+
+#ifdef CONFIG_USERSPACE
+void llext_entry(void *arg0, void *arg1, void *arg2)
+{
+	void (*fn)(void) = arg0;
+
+	LOG_INF("calling fn %p from thread %p", fn, k_current_get());
+	fn();
+}
+#endif /* CONFIG_USERSPACE */
+
+void load_call_unload(struct llext_test *test_case)
+{
+	struct llext_buf_loader buf_loader =
+		LLEXT_BUF_LOADER(test_case->buf, test_case->buf_len);
+	struct llext_loader *loader = &buf_loader.loader;
+	struct llext_load_param ldr_parm = LLEXT_LOAD_PARAM_DEFAULT;
+	struct llext *ext = NULL;
+
+	int res = llext_load(loader, test_case->name, &ext, &ldr_parm);
+
+	zassert_ok(res, "load should succeed");
+
+	void (*test_entry_fn)() = llext_find_sym(&ext->exp_tab, "test_entry");
+
+	zassert_not_null(test_entry_fn, "test_entry should be an exported symbol");
+
+#ifdef CONFIG_USERSPACE
+	/*
+	 * Due to the number of MPU regions on some parts with MPU (USERSPACE)
+	 * enabled we need to always call into the extension from a new dedicated
+	 * thread to avoid running out of MPU regions on some parts.
+	 *
+	 * This is part dependent behavior and certainly on MMU capable parts
+	 * this should not be needed! This test however is here to be generic
+	 * across as many parts as possible.
+	 */
+	struct k_mem_domain domain;
+
+	k_mem_domain_init(&domain, 0, NULL);
+
```

Comment:
This minimal domain setup causes a crash on `mps2_an521:llext.simple.readonly_mpu` because the thread is missing access to the libc partition (when there is one). The fix is to include `<zephyr/sys/libc-hooks.h>` and add the following code here:

```suggestion
#if Z_LIBC_PARTITION_EXISTS
	/* Include the z_libc partition in the domain */
	k_mem_domain_add_partition(&domain, &z_libc_partition);
#endif

```